### DISCUSSION TITLE
Discussão: publicação de arquivos SASS/LESS quando realizar um deploy.

### DISCUSSION THREAD

**Discussion by juliovedovatto** [frontendbr/forum#1361] [Discussão]
Pessoal, uma discussão aqui:

*Vocês fazem deploy de arquivos `SASS`/`LESS` junto com o deploy normal?* Ou só guardam eles no repositório e não o publicam junto?

Por via de regra, não tem sentindo publicar um arquivo source (o bruto `.scss`, `.less`, etc) que vai ser transpilado pra `css`,
mas quero ver se já passaram por algum caso, se seguem essa regra.

Propus isso no meu ambiente de trabalho, mas não foi bem visto por alguns.

_PS: minha questão não é sobre o sourcemaps em si (acho eles importantes), mas enviar os arquivos brutos SASS/LESS pra prod._

**Comment by onlurking** [comment]
Agora mesmo está tendo uma discussão no repo do [Rails](https://github.com/rails/webpacker/issues/769) sobre shippar sourcemaps em produção.

> I've learned so much from View Source over the years. From structure, to CSS, to JS. That has gotten harder with minification and transpiling. That's a real shame. The web has given us so much. The default should be to give back, in all the ways we can. View Source is a heritage feature that a framework like Rails should go out of its way to pay tribute to.

O @sindresorhus também possui uma opinião bem forte sobre shippar [sourcemaps](https://twitter.com/sindresorhus/status/1089075390327316480) em prod.

Na minha opinião faz sentido incluir os sourcemaps porque o CSS resultante não é otimizado pra leitura, sim apenas pro browser renderizar(ás vezes os namespaces são gerados automaticamente pra evitar colisões, mas isso é relativamente inutil já que tu consegue evitar colisões com Scoped CSS e uma metodologia decente pra escrever CSS(como **BEM**)).

Ter sourcemaps facilita muito o debugging  quando algum problema ocorre em produção, e tu não precisa ficar quebrando a cabeça pra integrar(de forma anônima) em serviços de error tracking(como rollbar e sentry), fora que você contribui pra transparência da internet.

O único contra shippar sourcemaps é que seu site vai ser levemente mais pesado, mas de prós você tem todos esses pontos.

**Comment by juliovedovatto** [comment]
@onlurking boa resposta, porém não era sobre isso :P 

Eu editei meu tópico para melhorar o entendimento. 

Basicamente, no meu ambiente de trabalho, tem gente (fullstack) que defende enviar pra prod os arquivos `.scss`, `.less` nos quais trabalhamos. Eu já acho desnecessário e omito eles no deploy.

**Comment by williangd** [comment]
@juliovedovatto mas qual seria o motivo, apontando pelo pessoal que defende o envio dos arquivos para produção? Eles serão usados para algo?

**Comment by juliovedovatto** [comment]
@williangd às vezes, por pressa e/ou pressão do cliente/gerente, tem dev que esquece de botar no git, mas envia manualmente pro site via FTP.

Tem casos que vai precisar baixar o que tem em produção pra atualizar o repositório git, dae um arquivo velho `.scss` velho (que ficou abandonado no servidor) pode sobrepor o que tem no git, dae diverge o layout. O dev nem vê a faz o commit sem o review, por pressa. E quando vê o CSS deu rollback e continuaram a trabalhar nele e nem perceberam.

Dae defendem deixar lá o arquivo raw do CSS, pra evitar coisas assim, pq cliente pode mexer, outro dev desconhecido pode mexer.

Usamos o padrão de não publicar o `.scss` faz 4 meses, porém nem todos  da equipe estavam ciente disto e não gostaram quando souberam e foi levantada toda uma discussão.

**Comment by evertonthepaula** [comment]
Então sua pergunta não é se deve ou não mandar os arquivos de desenvolvimento para o ambiente de produção, mas sim se todo o processo de desenvolvimento está correto. Pelo o que percebi vocês tem problemas sérios na gestão do produto. Mas respondendo sua pergunta, o ambiente de produção deve ser o mais enxuto possível, e é por isso que em teoria você deveria ter ambientes de: desenvolvimento, testes, homologação e produção. SE o cliente trocar de empresa desenvolvedora, aí é uma questão legal de com quem vai ficar o fonte.

Edit2: Pra que versionamento se vcs estão pegando o código direto do servidor? o_0

Edit1: Em tempo, não estou reprovando seu método, só estou dizendo que você está fazendo a pergunta errada.

**Comment by vNNi** [comment]
Acredito que não deve ter nenhum SASS/LESS em prod, até porque produção é só o que o browser precisa para rodar seu app e não algo legível/editável, tanto é que _minificamos_ o js, css;
Concordo com o @evertonthepaula, acho que você tem que olhar pro processo de desenvolvimento em si.

**Comment by tarciozemel** [comment]
> Vocês fazem deploy de arquivos `SASS`/`LESS` junto com o deploy normal?

Não.

> Ou só guardam eles no repositório e não o publicam junto?

Isso mesmo.

> Não tem sentindo publicar um arquivo source (o bruto `.scss`, `.less`, etc) que vai ser transpilado pra `css`

Exatamente.
